docs/src/main/asciidoc/mongo-guide.adoc

```diff
@@ -0,0 +1,599 @@
+        }
+        return null;
+    }
+
+}
+----
+
+Quarkus takes care of registering the `CodecProvider` for you.
+
+Finally, when getting the `MongoCollection` from the database you can use directly the `Fruit` class instead of the `Document` one, the codec will automatically map the `Document` to/from your `Fruit` class.
+
+Here is an example of using a `MongoCollection` with the `FruitCodec`.
+
+[source,java]
+----
+package org.acme.rest.json;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+
+@ApplicationScoped
+public class CodecFruitService {
+
+    @Inject MongoClient mongoClient;
+
+    public List<Fruit> list(){
+        List<Fruit> list = new ArrayList<>();
+        MongoCursor<Fruit> cursor = getCollection().find().iterator();
+
+        try {
+            while (cursor.hasNext()) {
+                list.add(cursor.next());
+            }
+        } finally {
+            cursor.close();
+        }
+        return list;
+    }
+
+    public void add(Fruit fruit){
+        getCollection().insertOne(fruit);
+    }
+
+    private MongoCollection<Fruit> getCollection(){
+        return mongoClient.getDatabase("fruit").getCollection("fruit", Fruit.class);
+    }
+}
+----
+
+
+== Building a native executable
+
+You can use the Mongo client in a native executable.
+
+If you want to use SSL/TLS encryption, you need to add these properties in your `application.properties`:
+----
+quarkus.mongodb.tls=true
+quarkus.mongodb.tls-insecure=true #only if TLS certificat cannot be validated
+----
+
+You can then build a native executable with the usual command `./mvnw package -Pnative`.
+
+Running it is as simple as executing `./target/mongo-client-1.0-SNAPSHOT-runner`.
+
+You can then point your browser to `http://localhost:8080/fruits.html` and use your application.
+
+[WARNING]
+====
+Currently, Quarkus doesn't support the `mongodb+srv` protocol in native mode.
+====
+
+== Conclusion
+
+Accessing a Mongo database from a Mongo Client is easy with Quarkus as it provides configuration and native support for it.
+
+== Connection properties
+`quarkus.mongodb.connection-string`:: (e.g. `mongodb://mongo1:27017,mongo2:27017`)
+MongoDB connection string, this is a URI where a lot of options can be passed : https://docs.mongodb.com/manual/reference/connection-string/
+
+`quarkus.mongodb.hosts`::
+Coma separated lits of hosts.
+
+`quarkus.mongodb.application-name`::
+Application name.
+
+`quarkus.mongodb.max-pool-size`::
+Maximum number of connections in the connection pool.
+
+`quarkus.mongodb.min-pool-size`::
+Minumum number of connections in the connection pool.
+
+`quarkus.mongodb.max-connection-idle-time`::
+Maximum idle time in milliseconds of a pooled connection. A connection that exceeds this limit will be closed.
```

Comment:
Should they be `Duration` instead? Same question for all these variables.

(we have a note to include if this is the case)

Reply:
Wondering about it. This is the MongClient configuration passed almost directly to the configuration. But yes, I would prefer using duration.

Reply:
Fixed - used duration everywhere.